### DISCUSSION TITLE
Group classes under a modifier

### DISCUSSION THREAD

**Discussion by unlocomqx** [tailwindlabs/tailwindcss#6312] [Ideas]
Hi,

I'm sure this is a duplicate because it looks kinda obvious (I searched but didn't find a similar discussion)

Instead of repeating a modifier many times, why not regroup the classes
Example
```html
 <input type="file" class="file:mr-6 file:px-4 file:py-2"/>
```
becomes something like
```html
<input type="file" class="file:(mr-6|px-4|py-2)"/>
```

This could be useful especially for more complex modifiers like `lg:dark:hover:...`

wdyt?
Thanks

**Comment by billriess** [comment]
If I recall correctly, space and comma are invalid characters in the spec.

**Reply by unlocomqx** [reply]
I forgot that the class should be compatible with the browser
I suggest maybe
```html
<input type="file" class="file:(mr-6|px-4|py-2)"/>
```

**Reply by billriess** [reply]
IF this were to be added, I think it would follow the `_` syntax used for arbitrary properties in Tailwind 3.0.

**Comment by JonathanFager** [comment]
I'd really like this but my prefered style would be

```html
<input type="file" class="file(mr-6 px-4 py-2) />
```

Without putting too much thought into it, I see two ways of stacking.

```html
<input type="file" class="file(mr-6 px-4 py-2 hover(bg-red-200)) />
```


or


```html
<input type="file" class="file(mr-6 px-4 py-2) file:hover(bg-pink-300) />
```


I would suggest this also apply to breakpoints. I've found myself having real issue of keeping track of my styles when styling for mobile, tablet, laptop, desktop with having 4 slight variations of the same style.

```html
<div class="flex flex-col md(flex-row)>
```

**Reply by unlocomqx** [reply]
I forgot that you can't have spaces in the class name, so I changed my suggestion.
Your idea is great too imo
It can become like this
```html
<input type="file" class="file(mr-6|px-4|py-2|hover(bg-red-200)) />
```

**Comment by cjayyy** [comment]
Isn't there any solution to this by now? 🙏

**Reply by wongjn** [reply]
There have been suggestions about adding such a feature. The maintainers have addressed this request before, that you may wish to read up on if you have not already done so:
- [A Twitter thread from the creator of Tailwind, Adam Wathan](https://twitter.com/adamwathan/status/1461519820411789314)
- [A previous GitHub discussion thread with maintainer responses](https://github.com/tailwindlabs/tailwindcss/discussions/8337#discussioncomment-4032611)

**Reply by cjayyy** [reply]
Thank you for the redirection, that pretty much sums it up. Hopefully soon the performance issue will get solved somehow. 🤞
